src/theme/themeManager.ts

```diff
@@ -0,0 +1,58 @@
+export type CSSModuleClassNames = {
+	[key: string]: boolean;
+}
+
+export type AppliedClasses<T> = {
+	[P in keyof T]?: CSSModuleClassNames;
+}
+
+type Theme = {
+	[key: string]: string;
+}
+
+type CacheKey = {
+	baseThemeClasses: {};
+	overrideClasses?: {};
+};
+
+let currentTheme: Theme = {};
+let cacheMap = new Map<CacheKey, AppliedClasses<any>>();
+
+function addClassNameToMap(classMap: CSSModuleClassNames, classList: Theme, className: string) {
+	if (classList && classList.hasOwnProperty(className)) {
```

Comment:
This can also be generic I think `addClassNameToMap<T extends Theme>(classMap: CSSModuleClassNames, classList: T, className: string)`? but perhaps that doesn't add anything though....